src/services/auth/plugins/password/index.ts

```diff
@@ -1,93 +1,67 @@
+import { LOGIN_TOKEN_EXPIRATION_IN_MINUTES, PUBLIC_URL } from '../../../../utils/config';
 import { buildRepositories } from '../../../../utils/repositories';
 import { getRedirectionUrl } from '../../utils';
+import captchaPreHandler from '../captcha';
+import { authenticatePassword, authenticatePasswordReset, authenticated } from '../passport';
 import {
   passwordLogin,
   patchResetPasswordRequest,
   postResetPasswordRequest,
   updatePassword,
 } from './schemas';
-import { MemberPasswordService } from './service';
-
-const PASSPORT_STATEGY_ID = 'jwt-reset-password';
 
 const plugin: FastifyPluginAsync = async (fastify) => {
-  const { mailer, log, db, redis } = fastify;
-
-  await fastify.register(fastifyPassport.initialize());
-  await fastify.register(fastifyPassport.secureSession());
-
-  fastifyPassport.use(
-    PASSPORT_STATEGY_ID,
-    new Strategy(
-      {
-        jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken(),
-        secretOrKey: PASSWORD_RESET_JWT_SECRET,
-        passReqToCallback: true,
-      },
-      async (req, payload, verifiedCallback) => {
-        if (payload.uuid && (await memberPasswordService.validatePasswordResetUuid(payload.uuid))) {
-          // Token has been validated
-          // Error is null, req.user is the Password Reset Request UUID.
-          const user: PassportUser = { uuid: payload.uuid };
-          return verifiedCallback(null, user);
-        } else {
-          // Authentication refused
-          // Error is null, user is false to trigger a 401 Unauthorized.
-          return verifiedCallback(null, false);
-        }
-      },
-    ),
-  );
-  // Register user object to session
-  fastifyPassport.registerUserSerializer(async (user: PassportUser, _req) => user.uuid);
-
-  const memberPasswordService = new MemberPasswordService(mailer, log, redis);
-
+  const {
+    db,
+    memberPassword: { service: memberPasswordService },
+  } = fastify;
   // login with password
   fastify.post<{
     Body: { email: string; password: string; captcha: string; url?: string };
-  }>('/login-password', { schema: passwordLogin }, async (request, reply) => {
-    const { body, log } = request;
-    const { url } = body;
-
-    // validate captcha
-    await fastify.validateCaptcha(request, body.captcha, RecaptchaAction.SignInWithPassword, {
-      shouldFail: false,
-    });
-
-    const token = await memberPasswordService.login(undefined, buildRepositories(), body);
-    const redirectionUrl = getRedirectionUrl(log, url);
+  }>(
+    '/login-password',
+    {
+      schema: passwordLogin,
+      preHandler: [
+        captchaPreHandler(RecaptchaAction.SignInWithPassword, {
+          shouldFail: false,
+        }),
+        authenticatePassword,
+      ],
+    },
+    async (request, reply) => {
+      const { body, log, user } = request;
+      const { url } = body;
+      const token = await memberPasswordService.generateToken(
+        { sub: user!.member!.id },
```

Comment:
As we discussed with you (@Defferrard ) and @spaenleh this after noon, maybe it could be interesting to use a method that make the assert concerning a possible undefined attribute. I don't know what could be a good name for the function and for the exception, but it is what we talked about:

```
function assert<T>(o: T | undefined): T{
  if (!o) { 
    throw new UndefinedParameterException();
  }
  return o
}
```

```suggestion
        { sub: assert(user?.member?.id) },
```

This allow us to better explain the error comparing to the null assert of typescript (!).